PythonVisualizations/Heap.py

```diff
@@ -0,0 +1,518 @@
+import time
+import random
+from tkinter import *
+
+try:
+    from drawable import *
+    from VisualizationApp import *
+except ModuleNotFoundError:
+    from .drawable import *
+    from .VisualizationApp import *
+
+
+class Heap(VisualizationApp):
+    nextColor = 0
+    MAX_ARG_WIDTH = 8
+    MAX_SIZE = 31
+    CELL_WIDTH = 25
+    CELL_HEIGHT = 12
+    CELL_BORDER = 1
+    CELL_BORDER_COLOR = 'black'
+    HEAP_X0 = 50
+    HEAP_Y0 = 30
+
+    def __init__(self, size=2, maxArgWidth=MAX_ARG_WIDTH, title="Heap", **kwargs):
+        super().__init__(title=title, maxArgWidth=maxArgWidth, **kwargs)
+        self.size = size
+        self.title = title
+        self.list = []
+        self.maxArgWidth = maxArgWidth
+        self.buttons = self.makeButtons()
+        self.display()
+        self.indexDisplay = self.createIndex(len(self.list))

```

Comment:
The `nItems` index should be created in the `display()` method.  That's where all of the canvas items representing the data structure are built.  That will allow it to be present along with the rectangles representing the array cells.  The version of Heap.py in `origin/master` creates this as `self.nItemsIndex`.

Oddly, the `indexDisplay` created on line 32 disappears when the Random Fill operation is run.  That shouldn't happen.